src/ngAnimate/animateCss.js

```diff
@@ -665,7 +665,7 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
       // we need to recalculate the delay value since we used a pre-emptive negative
       // delay value and the delay value is required for the final event checking. This
       // property will ensure that this will happen after the RAF phase has passed.
-      if (timings.transitionDuration > 0) {
+      if (options.duration == null && timings.transitionDuration > 0) {
```

Comment:
The reason why we have `|| 0` is because getComputedStyle is mocked out in a lot of parts of the tests and it returns NaN when the computeTimings code is run. Normally (outside of the testing world) getComputedStyle will always return a value of `0s` for transition-delay or transition-duration or animation-delay or animation-duration so there is no need to check for null values. This is really just here to make the mocking in tests work.